lib/posix/sleep.c

```diff
@@ -14,8 +14,12 @@
  */
 unsigned sleep(unsigned int seconds)
```

Comment:
Should we test the `seconds`? We can't take `0xffffff` because this has special meaning for Zephyr, `K_TICKS_FOREVER`.

```c
int32_t z_impl_k_sleep(k_timeout_t timeout)
{
	k_ticks_t ticks;

	__ASSERT(!arch_is_in_isr(), "");

	SYS_PORT_TRACING_FUNC_ENTER(k_thread, sleep, timeout);

	/* in case of K_FOREVER, we suspend */
	if (K_TIMEOUT_EQ(timeout, K_FOREVER)) {
		k_thread_suspend(_current);

		SYS_PORT_TRACING_FUNC_EXIT(k_thread, sleep, timeout, (int32_t) K_TICKS_FOREVER);

		return (int32_t) K_TICKS_FOREVER;
```

Reply:
I would be quite impressed to see a Zephyr system made today still running 136 years from now, at the very least or a system that could maintain power for that long..

Philosophically speaking though, if we waited *forever*, it would still be at least 2^32 - 1 seconds, so the POSIX spec is still satisfied